src/Native/Runtime/thread.h

```diff
@@ -135,7 +135,7 @@ class Thread : private ThreadBuffer
     void ClearState(ThreadStateFlags flags);
     bool IsStateSet(ThreadStateFlags flags);
```

Comment:
I guess the reason why you needed to add all these tweaks is because the default calling convention is not set to stdcall on Windows i386. We need to add this to the top level cmakefile:

```
  if (CLR_CMAKE_PLATFORM_ARCH_I386)
    add_compile_options(/Gz)
  endif (CLR_CMAKE_PLATFORM_ARCH_I386)
```

stdcall calling convention is the standard that we use on Windows i386. (There are likely number of other places that need it  to work, but the compiler is not warning you about.)